peru/main.py

```diff
@@ -221,6 +221,31 @@ def do_override(params):
             print('{}: {}'.format(module, params.runtime.get_override(module)))
 
 
+@peru_command('module', '''\
+Usage:
+    peru module <list> [-z]
+
+Options:
+    -z              print one module per line for machine-readable formatting
+
+''')
+def do_list(params):
+    modules = params.scope.modules.values()
+    output = ''
+    delim = ' | ' if params.args['-z'] else "\n  "
+
+    for module in modules:
+        output += module.name
+        for field, val in module.plugin_fields.items():
+            output += '{}{}: {}'.format(delim, field, val)
+        if(params.runtime.get_override(module.name)):
+            output += '{}{}: {}'.format(delim, 'override', params.runtime.get_override(module.name))
+        output += '\n'
+
+    if (output):
+        params.runtime.display.print(output[:-1])
```

Comment:
It looks the formatting we're doing would work well with a string `join`, rather than appending a delimiter that we need to strip off at the end. (For example, with this approach, we'd have to remember to update this line if we ever switched to a delimiter that was more than one character long.)